tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -510,6 +523,12 @@ class ValidationTest : public ::testing::Test {
             pos += replaceLen;
           }
           if (!bOptional) {
+            if (!found) {
+              WEX::Logging::Log::Comment(WEX::Common::String().Format(
+                  L"String not found: '%S' in text:\r\n%.*S",
```

Comment:
I got sick and tired of stepping through code in the debugger to see which detail went wrong each time, especially when there were a whole set of replacements being performed.  This should save lots of time in the future :).